charts/hnc/templates/hnc-controller-manager-ha.yaml

```diff
@@ -0,0 +1,104 @@
+{{- if .Values.ha.enabled }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: {{ include "hnc.fullname" . }}-controller-manager-ha
+  namespace: {{ include "hnc.namespace" . }}
+spec:
+  replicas: 3
+  selector:
+    matchLabels:
+      control-plane: controller-manager-ha
+  template:
+    metadata:
+      annotations:
+        prometheus.io/scrape: "true"
+      labels:
+        control-plane: controller-manager-ha
+    spec:
+      containers:
+        - args:
+            {{- if .Values.hrq.enabled }}
+            - --enable-hrq
+            {{- end }}
+            {{- range $hncExcludeNamespace := .Values.hncExcludeNamespaces}}
+            - --excluded-namespace={{ $hncExcludeNamespace }}
+            {{- end }}
+            - --webhook-server-port=9443
+            - --metrics-addr=:8080
+            - --max-reconciles=10
+            - --apiserver-qps-throttle=50
+            - --nopropagation-label=cattle.io/creator=norman
+            - --webhooks-only
+          command:
+            - /manager
+          image: {{ .Values.image.repository }}:{{ .Values.image.tag | default "hnc-manager:latest" }}
+          livenessProbe:
+            failureThreshold: 1
+            httpGet:
+              path: /healthz
+              port: 8081
+            periodSeconds: 10
+          name: manager
+          ports:
+            - containerPort: 9443
+              name: webhook-server
+              protocol: TCP
+            - containerPort: 8080
+              name: metrics
+              protocol: TCP
+            - containerPort: 8081
+              name: healthz
+              protocol: TCP
+          readinessProbe:
+            httpGet:
+              path: /readyz
+              port: 8081
+            periodSeconds: 5
+          securityContext:
+            allowPrivilegeEscalation: false
+            capabilities:
+              drop:
+                - ALL
+            readOnlyRootFilesystem: true
+            runAsNonRoot: true
+            seccompProfile:
+              type: RuntimeDefault
+          startupProbe:
+            failureThreshold: 100
+            httpGet:
+              path: /readyz
+              port: 8081
+            periodSeconds: 5
+          volumeMounts:
+            - mountPath: /tmp/k8s-webhook-server/serving-certs
+              name: cert
+              readOnly: true
+          {{- with .Values.imagePullPolicy }}
+          imagePullPolicy: {{ .Values.image.imagePullPolicy }}
+          {{- end }}
+          {{- with .Values.ha.manager.resources }}
+          resources: {{- toYaml . | nindent 12}}
+          {{- end }}
+      securityContext:
+        fsGroup: 2000
+        runAsNonRoot: true
+        runAsUser: 1000
+      terminationGracePeriodSeconds: 10
+      volumes:
+        - name: cert
+          secret:
+            defaultMode: 420
+            secretName: hnc-webhook-server-cert
```

Comment:
Thank you so much for working on this, and sorry in advance for butting in the PR (not a maintainer/reviewer of this project).

While testing your chart I noticed that this config is hard-coded and does not match the secret template name.

Also the secret template would most likely benefit from having `pre-install,pre-upgrade` helm hook added to make sure it's created before the controller to avoid race condition when installing/upgrading